docs/_docs/interface-manual.md

```diff
@@ -0,0 +1,11 @@
+# The Sampling Interface
+
+Turing's sampling interface presents several structures and functions that one needs to overload in order to implement an interface-compatible sampler. Currently it is non-operational, but the intent is to have all of Turing's samplers using the API below.
+
+1. A subtype of `AbstractSampler`, defined as a mutable struct containing state information
+2. A subtype of `AbstractTransition`, which represents a single draw from the sampler
+3. A function `transition_type` which returns the `AbstractTransition` type used by an implementation of an `AbstractSampler`
+4. A function `sample_init!` which performs any necessary set up
+5. A function `step!` which returns an `AbstractTransition`
```

Comment:
What's the mutated part here? Maybe clarify.

Reply:
This whole document is currently being reworked to include a tutorial on implementing MH, and I will clarify this. For your knowledge, any mutating function is meant to be as general as possible -- you might use 

- `sample_init!` to run an adaptation cycle (which in our case mutates the `Sampler`)
- `step!` might mutate a sampler flag after each sample. MH does this for example by using a `violating_support` flag.
- `sample_end!` contains any wrap-up you might need to do. If you were sampling in a transformed space, this might be where you convert everything back to a constrained space.